testbed/coav-sim.sh

```diff
@@ -41,11 +46,31 @@ LOGDIR=${LOGDIR:-"${SCRIPT_DIR}/logs"}
 GZSITL_UDP_PORT=15556
 COAV_GCS_UDP_PORT=15557
 
+run_and_log() {
+    cmd=$1
+    log_out=$2
+    log_err=$3
+    mode=$4
+
+    # Log to file
+    if [ "$VERBOSE_LEVEL" -eq 0 ]; then
+        $cmd > "${LOGDIR}/${log_out}" 2> "${LOGDIR}/${log_err}" &
+    # Log to file and stdout
+    elif [ "$VERBOSE_LEVEL" -eq 1 ]; then
+        $cmd > >(tee "${LOGDIR}/${log_out}") 2> >(tee "${LOGDIR}/${log_err}") &
+    # Log to stdout only
+    else
+        $cmd &
+    fi
+}
+
 run_autopilot () {
     # Run SITL Simulator
     if [ "$AUTOPILOT" = "AP_PX4" ]; then
         cd $PX4_DIR
-        $PX4_CMD > ${LOGDIR}/sitl.log 2> ${LOGDIR}/sitl_err.log &
+
+        run_and_log "$PX4_CMD" "sitl.log" "sitl_err.log"
```

Comment:
Maybe you could use just a log identifier and append ".log" and "_err.log" inside 'run_and_log'